test/scheduler.test.ts

```diff
@@ -130,4 +130,237 @@ describe("Schedule Store", () => {
+        {},
+      ),
+    ).rejects.toThrowError("Already exists");
+  });
+
+  test("Schedule Store: Get schedule that exists", async () => {
+    const scheduleId = "existing-schedule";
+    const cronExpression = "* * * * *"; // Every minute
+
+    // Create the schedule
+    await store.schedules.create(
+      scheduleId,
+      scheduleId,
+      "Existing Schedule",
+      cronExpression,
+      {},
+      "promise-1",
+      60000,
+      {},
+      '{"message": "Test"}',
+      {},
+    );
+
+    // Get the existing schedule
+    const existingSchedule = await store.schedules.get(scheduleId);
+    expect(existingSchedule.id).toBe(scheduleId);
+  });
+
+  test("Schedule Store: Get schedule that does not exist", async () => {
+    const nonExistingScheduleId = "non-existing-schedule-id";
+
+    // Attempt to get a schedule that does not exist, should throw NOT_FOUND error
+    await expect(store.schedules.get(nonExistingScheduleId)).rejects.toThrowError("Not found");
+  });
+
+  test("Schedule Store: Delete schedule that exists", async () => {
+    const scheduleId = "schedule-to-delete";
+
+    // Create the schedule first
+    await store.schedules.create(
+      scheduleId,
+      scheduleId,
+      "Existing Schedule",
+      "* * * * *",
+      {},
+      "promise-1",
+      60000,
+      {},
+      '{"message": "Test"}',
+      {},
+    );
+
+    // Attempt to delete the schedule
+    const isDeleted = await store.schedules.delete(scheduleId);
+    expect(isDeleted).toBeUndefined();
+
+    // Attempt to get the deleted schedule, should throw NOT_FOUND error
+    await expect(store.schedules.get(scheduleId)).rejects.toThrowError("Not found");
+  });
+
+  test("Schedule Store: Delete schedule that does not exist", async () => {
+    const nonExistingScheduleId = "non-existing-schedule-id";
+
+    // Attempt to delete a schedule that does not exist, should throw NOT_FOUND error
+    await expect(store.schedules.delete(nonExistingScheduleId)).rejects.toThrowError("Not found");
+  });
+  test("Schedule Store: Search by id", async () => {
+    const scheduleId = "search-by-id-schedule";
+
+    // Create the schedule
+    await store.schedules.create(
+      scheduleId,
+      scheduleId,
+      "Search by ID Schedule",
+      "* * * * *",
+      {},
+      "promise-1",
+      60000,
+      {},
+      '{"message": "Test"}',
+      {},
+    );
+
+    // Search for the schedule by id
+    let schedules: Schedule[] = [];
+    for await (const searchResults of store.schedules.search(scheduleId, {})) {
+      schedules = schedules.concat(searchResults);
+    }
+    expect(schedules.length).toBe(1);
+    expect(schedules[0].id).toBe(scheduleId);
+  });
+
+  test("Schedule Store: Search by id with wildcard(s)", async () => {
+    const scheduleIdPrefix = "search-by-id-prefix";
+    const wildcardSearch = `${scheduleIdPrefix}*`;
+
+    // Create multiple schedules with the same prefix
+    for (let i = 1; i <= 3; i++) {
```

Comment:
Can we create a few promises that do not match? One way to do it would be to loop over an array of schedule names, that way we have an array we can use to delete the schedules as well.
```ts
const schedules = [
 "should-match-1",
 "should-match-2",
 "should-match-3",
 "should-not-match-1",
 "should-not-match-2",
 "should-not-match-3",
];
```